themes/theme-boilerplate/views/default.master.tpl

```diff
@@ -34,14 +34,15 @@
             </div>
         {/if}
     {/if}
-    {categories_link format=$linkFormat}
-    {discussions_link format=$linkFormat}
-    {if $DataDrivenTitleBar}
-        {knowledge_link format=$linkFormat}
-    {/if}
+    
     {if !$DataDrivenTitleBar}
         {activity_link format=$linkFormat}
+        {categories_link format=$linkFormat}
+        {discussions_link format=$linkFormat}
+        {knowledge_link format=$linkFormat}
```

Comment:
I am not sure, if `knowledge_link` is `null` will it output anything here?

Reply:
It will output empty string https://github.com/vanilla/vanilla/blob/31bf81880c9c91c15b5f4f8f9ac9fe5962ca9346/library/SmartyPlugins/function.knowledge_link.php#L16

Unfortunately we can't conditionally define smarty functions in our current system or we get a fatal error.